File: 프로그래머스/Lv.0/안전지대/120866.java
class Solution {
    public int solution(int[][] board) {
        int answer = 0;
        
        int[][] board2 = new int[board.length][board[0].length];
        for (int i=0; i<board.length; i++) {
            for (int j=0; j<board[0].length; j++) {
                if (board[i][j] == 1) {
                    board2[i][j] = -1;
                    if (i - 1 >= 0) {
                        board2[i-1][j] = -1;
                        if (j - 1 >= 0) board2[i-1][j-1] = -1;
                        if (j + 1 < board[0].length) board2[i-1][j+1] = -1;
                    }
                    if (j - 1 >= 0) {
                        board2[i][j-1] = -1;
                        if (i - 1 >= 0) board2[i-1][j-1] = -1;
                        if (i + 1 < board.length) board2[i+1][j-1] = -1;
                    }
                    if (i + 1 < board.length) {
                        board2[i+1][j] = -1;
                        if (j - 1 >= 0) board2[i+1][j-1] = -1;
                        if (j + 1 < board[0].length) board2[i+1][j+1] = -1;
                    }
                    if (j + 1< board[0].length) {
                        board2[i][j+1] = -1;
                        if (i - 1 >= 0) board2[i-1][j+1] = -1;
                        if (i + 1 < board.length) board2[i+1][j+1] = -1;
                    }
                }
            }
        }
        
        for (int a=0; a<board.length; a++) {
            for (int b=0; b<board[0].length; b++) {
                if (board2[a][b] == 0) answer++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/Lv.0/안전지대/120866.js
function solution(board) {
    let answer = 0;
    
    const n = board.length;
    const board2 = Array.from({ length: n}, () => Array(n).fill(0));
    
    const dx = [-1, -1, -1, 0, 0, 1, 1, 1];
    const dy = [-1, 0, 1, -1, 1, -1, 0, 1];
    
    for (let i=0; i<n; i++) {
        for (let j=0; j<n; j++) {
            if (board[i][j] === 1) {
                board2[i][j] = -1;
                
                for (let d=0; d<8; d++) {
                    const ni = i + dx[d];
                    const nj = j + dy[d];
                    if (ni >= 0 && ni < n && nj >= 0 && nj < n) {
                        board2[ni][nj] = -1;
                    }
                }
            }
        }
    }
    
    for (let i=0; i<n; i++) {
        for (let j=0; j<n; j++) {
            if (board2[i][j] === 0) answer++;
        }
    }
    
    return answer;
}
